### PR TITLE
Declare by death now  leads to discharge of patient in Consultation form

### LINKED ISSUE #4565: Modify behavior of Declare death option in patient consultation form
follow-up #4497 

image

- [ ] when we select " Declare death " in the consultation form, a mandatory field should be coming below it, to enter the cause of death 

image

- [ ] once "Declare death" consultation form is created, the consultation page, should be behavior similar to a patient discharge from care as expired

### PR DESCRIPTION
**Pull request by patelaryan7751**
## Proposed Changes

- Fixes #4565 ?
- Declare by death now  leads to discharge of patient in Consultation form
image
image

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

File: src/Components/Facility/ConsultationForm.tsx [hunk 1]
```diff
@@ -695,6 +759,39 @@ export const ConsultationForm = (props: any) => {
...
         )}
 
+        {state.form.suggestion === "DD" && (
+          <>
+            <div id="cause_of_death">
+              <TextAreaFormField
+                {...field("cause_of_death")}
+                required={state.form.suggestion === "DD"}
+                label="Cause of Death"
+                name="cause_of_death"
```
Comment (rithviknishad) [inline src/Components/Facility/ConsultationForm.tsx]: `name` is already set by `field(...)`

```suggestion
```

File: src/Components/Facility/ConsultationForm.tsx [hunk 2]
```diff
@@ -695,6 +759,39 @@ export const ConsultationForm = (props: any) => {
...
+                value={state.form.cause_of_death}
+              />
+            </div>
+            <div id="death_datetime">
+              <TextFormField
+                {...field("death_datetime")}
+                type="datetime-local"
+                required={state.form.suggestion === "DD"}
+                label="Date & Time of Death"
+                name="death_datetime"
```
Comment (rithviknishad) [inline src/Components/Facility/ConsultationForm.tsx]:
```suggestion
```

File: src/Components/Facility/ConsultationForm.tsx [hunk 3]
```diff
@@ -695,6 +759,39 @@ export const ConsultationForm = (props: any) => {
...
+                value={state.form.death_datetime}
+              />
+            </div>
+            <div id="death_confirmed_doctor">
+              <TextAreaFormField
+                {...field("death_confirmed_doctor")}
+                required={state.form.suggestion === "DD"}
+                label="Death Confirmed by"
+                name="death_confirmed_doctor"
```
Comment (rithviknishad) [inline src/Components/Facility/ConsultationForm.tsx]:
```suggestion
```

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**